jira_scraper/core/scraper.py

```diff
@@ -90,36 +106,57 @@ def fetch_all_issues(self, query: str, max_results: int) -> List[Dict]:
             issue for batch in results for issue in batch]
         return all_issues
 
-    def process_issues_to_dataframe(self, issues: List[Dict]) -> pd.DataFrame:
-        """Convert JIRA issues to pandas DataFrame."""
-        dataset = []
+    def get_jira_records(self, issues: List[Dict]) -> list[JiraRecord]:
+        """Convert Jira API responses to JiraRecords"""
+        jira_records: list[JiraRecord] = []
 
         for issue in tqdm(issues, desc="Processing issues"):
-
             jira_url = f"{self.config['jira_url']}/browse/{issue['key']}"
-            for text_field in ["summary", "description"]:
-                self.insert_row(
-                    issue["fields"][text_field],
-                    issue["id"],
-                    jira_url,
-                    text_field,
-                    dataset)
+
+            components = [
+                component["name"]
+                for component in issue["fields"]["components"]
+            ]
+
+            fix_versions = [
+                fixVersion["name"]
+                for fixVersion in issue["fields"]["fixVersions"]
+            ]
+
+            versions = [
+                version["name"]
+                for version in issue["fields"]["versions"]
+            ]
+
+            for text_field_name in ["summary", "description"]:
+                jira_records.append({
+                    "text": issue["fields"][text_field_name],
+                    "id": issue["id"],
+                    "affects_versions": versions,
+                    "components": components,
+                    "kind": text_field_name,
+                    "fix_versions": fix_versions,
+                    "url": jira_url,
+                    "key": issue["key"],
```

Comment:
Looks like we've introduced the `key` field again. I had dropped it in an earlier PR as the `url` has the issue key and it's unique.

Reply:
True. No need to store the info twice. Thanks! Let me fix this in a follow-up PR.

Reply:
https://github.com/RCAccelerator/tools/pull/22